packages/js/components/src/section/section.tsx

```diff
@@ -27,6 +32,9 @@ export function Section( { component, children, ...props } ) {
 					{ component === false ? (
 						children
 					) : (
+						// @ts-ignore: Raises JSX component `Component` doesn't have constructor to call error.
+						// Suppressing the error since this component accepts func, string, and bool
+						// I wasn't able to find to correct type.
 						<Component { ...props }>{ children }</Component>
```

Comment:
I couldn't find a way to type it correctly to suppress the error message. I'm open to any suggestion 🙏

Reply:
I think the **component** prop type should be `React.ComponentType | string | false;` since "true" is not invalid here. Otherwise, we can explicitly throw an error if `Component === true` for backward compatibility.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/36298/commits/3db25b3236f69ce73955240d75bce45398e411ec